pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -72,8 +72,9 @@ func (r *AtlasProjectReconciler) Reconcile(context context.Context, req ctrl.Req
 		return result.ReconcileResult(), nil
 	}
 	if project.ConnectionSecretObjectKey() != nil {
+		// Note, that we are not watching the global connection secret - seems there is no point in reconciling all
```

Comment:
[q] what if the `orgId` was changed? do we even support that case?

Reply:
yes - in this case the new project will be created in that organization and the stale project will stay in the old organization until it's removed manually

Reply:
what I mean is - don't we need to reconcile in this case?

Reply:
this is a good thing to discuss!
If say the customer has 5 projects in the organization each with 2 clusters. What would be the motivation to change the  `orgID` and to copy all the projects (and eventually all the clusters) to the new organization? If they want the organization with a new name - then the name of organization can be edited directly in Atlas. Apart from that  automatic creation of 5 projects in the new organization (without automatic creation of clusters though) right after the global API key changes doesn't look very user-friendly to me... If we don't create the new project right away - the old one (with the same name and IP configs) with all the clusters will still be alive

That being said even watching the "project-level" secret doesn't look very important though it will affect smaller number of projects unlike the global one which may create a bunch of projects right away. So maybe we don't need to watch the project-level secret at all as well...

What do you think? Do you think we should watch the global secret as well?

Reply:
I agree that this is probably not something the user would want to do, so this is more of a curious question than it is a feature request

So yeah, most likely we don't need to change this, just wanted to hear your thoughts on it 🙂